contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol

```diff
@@ -0,0 +1,604 @@
+        // Allow to reimburse if funding of the round was unsuccessful.
+        if (!round.hasPaid[_answer]) {
+            reward = round.contributions[_beneficiary][_answer];
+        } else if (finalAnswer == 0 || !round.hasPaid[finalAnswer]) {
+            // Reimburse unspent fees proportionally if there is no winner and loser. Also applies to the situation where the ultimate winner didn't pay appeal fees fully.
+            reward = round.fundedAnswers.length > 1
+                ? (round.contributions[_beneficiary][_answer] * round.feeRewards) /
+                    (round.paidFees[round.fundedAnswers[0]] + round.paidFees[round.fundedAnswers[1]])
+                : 0;
+        } else if (finalAnswer == _answer) {
+            // Reward the winner.
+            reward = round.paidFees[_answer] > 0
+                ? (round.contributions[_beneficiary][_answer] * round.feeRewards) / round.paidFees[_answer]
+                : 0;
+        }
+
+        if (reward != 0) {
+            round.contributions[_beneficiary][_answer] = 0;
+            _beneficiary.transfer(reward);
+            emit Withdrawal(_arbitrationID, _round, _answer + 1, _beneficiary, reward);
+        }
+    }
+
+    /** @notice Allows to withdraw any reimbursable fees or rewards after the dispute gets solved for multiple ruling options (answers) at once.
+     *  @param _arbitrationID The ID of the arbitration.
+     *  @param _beneficiary The address that made contributions.
+     *  @param _round The round from which to withdraw.
+     *  @param _contributedTo Answers that received contributions from contributor.
+     */
+    function withdrawFeesAndRewardsForMultipleRulings(
+        uint256 _arbitrationID,
+        address payable _beneficiary,
+        uint256 _round,
+        uint256[] memory _contributedTo
+    ) public override {
+        for (uint256 contributionNumber = 0; contributionNumber < _contributedTo.length; contributionNumber++) {
+            withdrawFeesAndRewards(_arbitrationID, _beneficiary, _round, _contributedTo[contributionNumber]);
+        }
+    }
+
+    /** @notice Allows to withdraw any rewards or reimbursable fees for multiple rulings options (answers) and for all rounds at once.
+     *  @dev This function is O(n) where n is the number of rounds. This could exceed gas limits, therefore this function should be used only as a utility and not be relied upon by other contracts.
+     *  @param _arbitrationID The ID of the arbitration.
+     *  @param _beneficiary The address that made contributions.
+     *  @param _contributedTo Answers that received contributions from contributor.
+     */
+    function withdrawFeesAndRewardsForAllRounds(
+        uint256 _arbitrationID,
+        address payable _beneficiary,
+        uint256[] memory _contributedTo
+    ) external override {
+        for (uint256 roundNumber = 0; roundNumber < arbitrations[_arbitrationID].rounds.length; roundNumber++) {
```

Comment:
Same as above. This is actually `O(n*m)`. Of course most of the times the beneficiary will contribute to a single answer, but the asymptotic complexity is not a function of specific parameters.